Initial support for $HOME and some XDG variables inside the configuration file

JoanVC100 opened the pull request:
This pull request tries to address issues #263 and #476.

With these changes you can use $HOME and some of the $XDG variables directly in the MPD configuration file. I used the functions in "fs/glue/StandardDirectory.hxx" to extract the XDG variables, so currently it's limited to:
- XDG_CONFIG_HOME
- XDG_MUSIC_DIR
- XDG_CACHE_HOME
- XDG_RUNTIME_DIR

MaxKellermann commented:
Your commit message is misleading - this is not about environment variables.
Also documentation is missing.

JoanVC100 commented:
Sorry for the premature closure, I am still learning about GitHub pull requests.

Anyway, I rewrote the commit message to make it more accurate. And I changed the documentation in both the man page and the HTML, as well as the example mpd.conf.

MaxKellermann commented:
Thanks, that's good!
